dartagnan/src/main/java/com/dat3m/dartagnan/Dartagnan.java

```diff
@@ -223,13 +197,119 @@ public static void main(String[] args) throws Exception {
+				}
+				final List<Axiom> violatedCATSpecs = task.getMemoryModel().getAxioms().stream()
+						.filter(Axiom::isFlagged)
+						.filter(ax -> FALSE.equals(model.evaluate(CAT_SPEC.getSMTVariable(ax, encCtx))))
+						.collect(Collectors.toList());
+				if (!violatedCATSpecs.isEmpty()) {
+					summary.append("CAT specification violation found").append("\n");
+					for (Axiom violatedAx : violatedCATSpecs) {
+						summary.append("Flag ")
+								.append(Optional.ofNullable(violatedAx.getName()).orElse(violatedAx.getNameOrTerm()))
+								.append("\n");
+					}
+				}
+			} else if (hasPositiveWitnesses) {
+				if(TRUE.equals(model.evaluate(PROGRAM_SPEC.getSMTVariable(encCtx)))) {
+					// The check above is just a sanity check: the program spec has to be true
+					// because it is the only property that got encoded.
+					summary.append("Program specification witness found.").append("\n");
+				}
+			}
+			summary.append(result).append("\n");
+		} else {
+			// Litmus-specific output format that matches with Herd7 (as good as it can)
+			if (p.getFilterSpecification() != null) {
+				summary.append("Filter ").append(p.getFilterSpecification().toStringWithType()).append("\n");
+			}
+
+			// NOTE: We cannot produce an output that matches herd7 when checking for both program spec and cat properties.
+			// This requires two SMT-queries because a single model is unlikely to witness/falsify both properties
+			// simultaneously.
+			// Instead, if we check for multiple safety properties, we find the first one and
+			// generate output based on its type (Program spec OR CAT property)
+
+			if (hasPositiveWitnesses) {
+				// We have a positive witness or no violations, then the program must be ok.
+				// NOTE: We also treat the UNKNOWN case as positive, assuming that
+				// looping litmus tests are unusual.
+				summary.append("Condition ").append(p.getSpecification().toStringWithType()).append("\n");
+				summary.append("Ok").append("\n");
+			} else if (hasViolations) {
+				if (FALSE.equals(model.evaluate(PROGRAM_SPEC.getSMTVariable(encCtx)))) {
+					// Program spec violated
+					summary.append("Condition ").append(p.getSpecification().toStringWithType()).append("\n");
+					summary.append("No").append("\n");
+				} else {
+					final List<Axiom> violatedCATSpecs = task.getMemoryModel().getAxioms().stream()
+							.filter(Axiom::isFlagged)
+							.filter(ax -> FALSE.equals(model.evaluate(CAT_SPEC.getSMTVariable(ax, encCtx))))
+							.collect(Collectors.toList());
+					for (Axiom violatedAx : violatedCATSpecs) {
+						summary.append("Flag ")
+								.append(Optional.ofNullable(violatedAx.getName()).orElse(violatedAx.getNameOrTerm()))
+								.append("\n");
+					}
+				}
+			} else {
+				// We have neither a witness nor a violation ...
+				if (result == UNKNOWN) {
+					// Sanity check: UNKNOWN is not an expected result for litmus tests
+					summary.append(result).append("\n");
```

Comment:
can we log a warning? I think this would be easier to spot for us that an UNKNOWN result when we are running litmus tests